rpc/websocket.go

```diff
@@ -112,60 +82,46 @@ func wsHandshakeValidator(allowedOrigins []string) func(*websocket.Config, *http
 			origins.Add(strings.ToLower(origin))
 		}
 	}
-
 	// allow localhost if no allowedOrigins are specified.
 	if len(origins.ToSlice()) == 0 {
 		origins.Add("http://localhost")
 		if hostname, err := os.Hostname(); err == nil {
 			origins.Add("http://" + strings.ToLower(hostname))
 		}
 	}
-
 	log.Debug(fmt.Sprintf("Allowed origin(s) for WS RPC interface %v", origins.ToSlice()))
 
-	f := func(cfg *websocket.Config, req *http.Request) error {
+	f := func(req *http.Request) bool {
 		// Skip origin verification if no Origin header is present. The origin check
 		// is supposed to protect against browser based attacks. Browsers always set
 		// Origin. Non-browser software can put anything in origin and checking it doesn't
 		// provide additional security.
 		if _, ok := req.Header["Origin"]; !ok {
```

Comment:
We need to use `req.Header.Get` here, otherwise the code won't ignore the case.

Reply:
That's actually not true. net/http canonicalizes header keys, i.e. the keys in the map are assumed to be in the form returned by [http.CanonicalHeaderKey](https://godoc.org/net/http#CanonicalHeaderKey).

Reply:
Using `req.Header.Get` is not possible because it doesn't distinguish between absent and empty headers.